src/todo_final.ts

```diff
@@ -84,21 +83,36 @@ function findTodoByText(text: string) {
 	);
 	return todoItems[findTodoWithText]; // return the item not the text itself
 }
+function findTodoById(id: number) {
+	// find the index where the  text that matches the text  inside the array
+	const findTodoWithId = todoItems.findIndex(
+		(todoInsideArray) => todoInsideArray.id === id
+	);
+	return todoItems[findTodoWithId]; // return the item not the text itself
+}
 
 // remove todo by text
 
 function removeTodoByText(text: string) {
 	return removeTodo(findTodoByText(text));
 }
+// remove todo by user Id
+function removeTodoById(id: number) {
+	return removeTodo(findTodoById(id)); //returns the return val of removeTodo
+}
 
-function removeTodo(itemsToRemove: any[number]) {
+function removeTodo(itemsToRemove: TodoItem) {
+	if (itemsToRemove === undefined) {
+		throw new Error("Item is undefined"); //class
+	}
 	// find the index of the todo with the id you are looking for and return index itself
+
 	const indexOfTodoToDelete = todoItems.findIndex(
-		(todoInArray: { id: any }) => todoInArray.id === itemsToRemove.id
+		(todoInArray: { id: any }) => todoInArray.id === itemsToRemove.id // can be undefined
 	);
 	// remove that todo
 	todoItems.splice(indexOfTodoToDelete, 1); // delete the todo
-	return itemsToRemove.id;
+	return itemsToRemove;
 }
```

Comment:
`itemsToRemove` shouldn't be plural here. There is only one item being removed, so the variable name should be `itemToRemove`.